freeipa-client/src/main/java/com/sequenceiq/freeipa/client/FreeIpaClient.java

```diff
@@ -485,6 +486,11 @@ public Host showHost(String fqdn) throws FreeIpaClientException {
         try {
             RPCResponse<T> response = (RPCResponse<T>) jsonRpcHttpClient.invoke(method, List.of(flags, parameterMap), type);
             LOGGER.debug("Response object: {}", response);
+            if (response == null) {
+                // TODO CDPCP-1028 investigate why invoke returns null instead of throwing an exception
+                // when the cluster-proxy request times out.
+                throw new NullPointerException("JSON-RPC response is null");
```

Comment:
Shouldn't we throw a freeipa client exception?

Reply:
it gets immediately wrapped and thrown as a FreeIpaClientException on line 498.